share/core.go

```diff
@@ -0,0 +1,315 @@
+// Package share implements different secret sharing mechanisms, including
+// simple Shamir secret sharing, verifiable secret sharing (vss), and public
+// verifiable secret sharing (pvss).
+package share
+
+import (
+	"crypto/cipher"
+	"errors"
+
+	"github.com/dedis/crypto/abstract"
+)
+
+// This file provides the core functionality for secret sharing including
+// Shamir's scheme and public commitment polynomials.
+
+// Some error definitions
+var errorGroups = errors.New("non-matching groups")
+var errorCoeffs = errors.New("different number of coefficients")
+
+// PriShare represents a private share.
+type PriShare struct {
+	I int             // Index of the private share
+	V abstract.Scalar // Value of the private share
+}
+
+// PriPoly represents a secret sharing polynomial.
+type PriPoly struct {
+	g      abstract.Group    // Cryptographic group
+	coeffs []abstract.Scalar // Coefficients of the polynomial
+}
+
+// NewPriPoly creates a new secret sharing polynomial for the cryptographic
+// group g, the secret sharing threshold t, and the secret to be shared s.
+func NewPriPoly(g abstract.Group, t int, s abstract.Scalar, rand cipher.Stream) *PriPoly {
+	coeffs := make([]abstract.Scalar, t)
+	coeffs[0] = s
+	if coeffs[0] == nil {
+		coeffs[0] = g.Scalar().Pick(rand)
+	}
+	for i := 1; i < t; i++ {
+		coeffs[i] = g.Scalar().Pick(rand)
+	}
+	return &PriPoly{g, coeffs}
+}
+
+// Threshold returns the secret sharing threshold.
+func (p *PriPoly) Threshold() int {
+	return len(p.coeffs)
+}
+
+// GetSecret returns the shared secret p(0), i.e., the constant term of the polynomial.
+func (p *PriPoly) GetSecret() abstract.Scalar {
```

Comment:
I'd rename that simply `Secret`. In fact I'm not even sure I'd put that as a function, because why would you need to get the secret when you are the one that built the pripoly...  but that's up to debate ;)